### ISSUE TITLE
Errors when using the web integration example with the lite data file - Could not find property 'Promise'/'Fetch'.

### ISSUE THREAD

**Issue by srdjo** [51Degrees/device-detection-dotnet#16]
Hi all, 

I am migrating from ASP.Net MVC 5 to Asp.Net Core 6 and I made an empty project just to test device detection because we use it a lot and I have a couple of issues.

First of them is this code in Program.cs

`builder.Services.AddSingleton<DeviceDetectionHashEngineBuilder>();
builder.Services.AddFiftyOne(builder.Configuration);`

After adding the last line, Hot reload feature stops working and I can not see any UI changes (.cshtml file) without rebuilding the application or restarting it.

Also on every page load I see this error in console

fail: FiftyOne.Pipeline.Core.FlowElements.Pipeline[0]
      Could not find property 'Promise'.
fail: FiftyOne.Pipeline.Core.FlowElements.Pipeline[0]
      Could not find property 'Fetch'.

![image](https://user-images.githubusercontent.com/4582491/141747659-3537a4af-8c6e-4893-b304-b92ad2590a4e.png)

I followed this example https://51degrees.com/device-detection-dotnet/4.3/_asp_net_core3_81_2_startup_8cs-example.html

What might be the problem ?

**Comment by Steve51D** [comment]
Hi @srdjo 

We've not tested against .NET 6 yet, or with hot reload specifically, so I'm not sure what might be happening there.
I'll raise this internally.

For the Promise/Fetch issue, could you confirm the version of the 51Degrees NuGet package(s) you're using? We had an identical problem that was fixed several months ago, so just want to eliminate that possibility.
Also, knowing the type and publish date of the data file you're using might be useful. (our [data file tool](https://github.com/51Degrees/data-file-tool) can tell you this info if you're not sure)

Thanks,
Steve.

**Comment by srdjo** [comment]
Hi @Steve51D ,

I am using 

FiftyOne.DeviceDetection - 4.3.9
FiftyOne.Pipeline.Web - 4.3.7

About Data file
Publish date: 31.10.2021

**Comment by Steve51D** [comment]
Thanks for the info.
The errors you are seeing are due to the ASP.NET integration trying to make use of properties that are not available in the lite data file. (`Promise` and `Fetch`)
It should just be falling back to less specific functionality though, rather than logging an error. 
We're not going to get a fix in time for the next release on 17th Nov, but we'll aim to get this looked at for a future release.

On the subject of .NET 6, we typically wait several months before starting to include tests against new language versions. This is just because our users are rarely on bleeding edge versions and waiting an extra few months to allow the release and ecosystem to mature a bit often smooths over rough spots in the initial release.

In the meantime, we welcome pull requests to add tests or address issues.
Also, If you particularly want tests to be added and conducted against a newer version of a language then you can contact our [commercial support](mailto:support@51degrees.com) team.

**Comment by Steve51D** [comment]
I've created a separate issue for adding .NET 6 tests here: #17 
We'll use this issue to track the fix for the bug re. the `Promise` and `Fetch` properties.

**Comment by Steve51D** [comment]
The issue with the errors being logged for the `Promise` and `Fetch` properties will be fixed in the next release on 1st Dec.
